src/modules/menu/components/share/MenuShareInputCopyButton.vue

```diff
@@ -1,17 +1,119 @@
+<script setup>
+import { computed, onBeforeUnmount, onMounted, onUpdated, ref, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+// import { useStore } from 'vuex'
+import { useTippyTooltip } from '@/utils/composables/useTippyTooltip'
+
+const props = defineProps({
+    inputText: {
+        type: String,
+        default: null,
+    },
+    small: {
+        type: Boolean,
+        default: true,
+    },
+    copyText: {
+        type: String,
+        default: 'copy_cta',
+    },
+    copiedText: {
+        type: String,
+        default: 'copy_done',
+    },
+    labelText: {
+        type: String,
+        default: null,
+    },
+    hasWarning: {
+        type: Boolean,
+        default: false,
+    },
+})
+
+const copiedInClipboard = ref(false)
+const timeoutCopied = ref(null)
+
+// const store = useStore()
+const i18n = useI18n()
+// const store = useStore()
+
+const { refreshTippyAttachment, removeTippy } = useTippyTooltip(
+    '#input-copy-button[data-tippy-content]',
+    {
+        placement: 'top',
+        offset: [0, -20],
+    }
+)
+
+const buttonText = computed(() => {
+    return i18n
+        .t(copiedInClipboard.value ? props.copiedText : props.copyText)
+        .replace('&nbsp;', '\xa0')
+})
+// const isPhoneMode = computed(() => store.getters.isPhoneMode)
+
+const clearIsCopiedInClipboard = () => {
+    copiedInClipboard.value = false
+    clearTimeout(timeoutCopied.value)
+}
+
+const copyInputToClipboard = () => {
+    navigator.clipboard.writeText(props.inputText)
+    copiedInClipboard.value = true
+    timeoutCopied.value = setTimeout(clearIsCopiedInClipboard, 2500)
+}
+
+watch(
+    () => props.inputText,
+    () => {
+        clearIsCopiedInClipboard()
+    }
+)
+
+onMounted(() => {
+    refreshTippyAttachment()
+})
+
+onBeforeUnmount(() => {
+    clearTimeout(timeoutCopied.value)
+})
+
+onUpdated(() => {
+    refreshTippyAttachment()
+    if (!props.hasWarning) {
+        removeTippy()
+    } else {
+        refreshTippyAttachment()
+    }
+})
+</script>
+
 <template>
-    <div v-if="inputText">
+    <div
+        v-if="inputText"
+        id="input-copy-button"
+        data-tippy-content="warn_share_local_file"
+        data-cy="input-copy-button"
+    >
         <label v-if="labelText">{{ $t(labelText) }}: </label>
         <div class="input-group" :class="{ 'input-group-sm': small }">
             <input
                 type="text"
                 class="form-control input-text-to-copy"
+                :class="{ 'warning-input': hasWarning }"
                 readonly="readonly"
                 :value="inputText"
                 data-cy="menu-share-input-copy-button"
                 @focus="$event.target.select()"
             />
             <button
-                class="btn btn-outline-group"
+                class="btn"
+                :class="{
+                    'warning-input': hasWarning,
```

Comment:
you should not use self-created class here to apply the color, but use `btn-warning` or such Bootstrap tools instead (`bg-warning`, etc...).

With the current way of doing it, the hover color is out of place (and I bet we can find some cases where the color won't be consistent by doing it ourselves)